### ISSUE TITLE
If I set state in using additionalParams object in authorizeUrl method then after checking response with exchangeForAccessToken I get error 'Invalid State' 

### ISSUE THREAD

**Issue by pauliusg** [bpedroza/js-pkce#13]
Example: 
```
...
const authorizeUrl = pkce.authorizeUrl({ state: 'myState' });
...
cons token = await pkce.exchangeForAccessToken(responseUrl);
```
Looks like if we want to validate state in such situation we need to set it under pkce_state in session storage.

**Comment by bpedroza** [comment]
State is currently managed (created/validated) internally so when you override it, what is expected does not match what you've sent. If you just want state to be sent and validated, it already is. I can see how this would be something you may want to set explicitly though. I will look into changing this behavior to handle this use case. Thank you for opening the issue.

**Comment by bpedroza** [comment]
I would recommend you not manually update session storage as a long term solution. There are plans to allow using other storage types, additionally there is no guarantee this package will continue to use that key for storing the state. If you need to do that in the short term until the package is updated, that should work.

**Comment by pauliusg** [comment]
@bpedroza , sure, I did workaroun only are temporary solution:
```
sessionStorage.setItem('pkce_state', myState);
```
And waiting for a fix, thank you.

**Comment by bpedroza** [comment]
This should be resolved in version 1.1.3. Please confirm it's working as expected for you now.

**Comment by pauliusg** [comment]
@bpedroza , thank you, working as expected.
